File: NITCMessManagerBackend/models/contractor.js
const mongoose = require('mongoose')
const menu = require('./menu')
const feedback = require('./feedback')
const config = require('config')
const jwt = require('jsonwebtoken')

const contractorSchema = new mongoose.Schema({
    contractorName: {type: 'string', required: true},
    contractorEmail: {type: 'string', required: true},
    contractorPassword: {type: 'string', required: true},
    messName: {type: 'string', required: true},
    foodType: {type: 'string'},
    capacity: {type: 'number'},
    availability: {type: 'number'},
    costPerDay: {type: 'number'},
    totalDue: {type: 'number', default: 0},
    userType: {type: 'string', required: true},
    studentEnrolled: [String],
    messMenu: [menu.schema],
    feedbackReceived: [feedback.schema]
})

contractorSchema.methods.generateAuthToken = function(){
    const jwtGenerated = jwt.sign({_id: this._id, email: this.email}, config.get('PRIVATE_KEY'))
    return jwtGenerated
}

module.exports = mongoose.model('contractor', contractorSchema)

File: NITCMessManagerFrontend/app/src/main/java/com/kumar/messmanager/model/Contractor.kt
package com.kumar.messmanager.model

import com.kumar.messmanager.model.Feedback
import com.kumar.messmanager.model.Menu
import com.kumar.messmanager.model.Student

data class Contractor(var _id : String = "",
                      var contractorEmail : String = "",
                      var contractorPassword : String = "",
                      var userType : String = "",
                      var contractorName : String = "",
                      var costPerDay : Int = 0,
                      var messName : String = "",
                      var foodType : String = "",
                      var capacity : Int = 0,
                      var availability : Int = 0,
                      var totalDue : Int = 0,
                      var totalEnrolled : Int = 0,
                      var messMenu : ArrayList<Menu> = ArrayList(),
                      var feedbackReceived : ArrayList<Feedback> = ArrayList(),
                      var studentEnrolled : ArrayList<String> = ArrayList()
) {
}